src/middlewared/middlewared/plugins/pool.py

```diff
@@ -2379,6 +2379,14 @@ def import_on_boot(self, job):
                     'aclmode': {'value': 'restricted'},
                 }})
 
+        try:
+            # Now that pools have been imported we are ready to configure system dataset
+            # and collectd which may depend on them.
+            self.middleware.call('etc.generate', 'system_dataset')
+            self.middleware.call('etc.generate', 'collectd')
```

Comment:
Is this intended to be skipped if `system_dataset` fails?

Reply:
Yes, there is no point in raising the exception and not continue on remaining tasks

Reply:
I mean, skip setting up collectd if system_dataset setup failed

Reply:
We probably need to prevent starting collectd if it is not set up

Reply:
If system dataset fails to the point of generating an exception we are probably already dead in the water and the system will malfunction